contracts/src/utils/MockPermissionedDisputeGame.sol

```diff
@@ -39,42 +66,41 @@ contract MockPermissionedDisputeGame is IDisputeGame {
         return _status;
     }
 
-    function gameType() external pure returns (GameType gameType_) {
+    function gameType() public pure returns (GameType gameType_) {
         gameType_ = GameType.wrap(1);
     }
 
-    function gameCreator() external pure returns (address creator_) {
-        creator_ = address(0);
+    function gameCreator() public pure returns (address creator_) {
+        creator_ = _getArgAddress(0x00);
     }
 
-    function rootClaim() external pure returns (Claim rootClaim_) {
-        rootClaim_ = Claim.wrap(bytes32(0));
+    function rootClaim() public pure returns (Claim rootClaim_) {
+        rootClaim_ = Claim.wrap(_getArgBytes32(0x14));
     }
 
-    function l1Head() external pure returns (Hash l1Head_) {
-        l1Head_ = Hash.wrap(bytes32(0));
+    function l1Head() public pure returns (Hash l1Head_) {
+        l1Head_ = Hash.wrap(_getArgBytes32(0x34));
     }
 
     function l2BlockNumber() external pure returns (uint256 l2BlockNumber_) {
-        l2BlockNumber_ = 0;
+        l2BlockNumber_ = _getArgUint256(0x54);
     }
 
-    function extraData() external pure returns (bytes memory extraData_) {
-        extraData_ = bytes("");
+    function parentIndex() public pure returns (uint32 parentIndex_) {
+        parentIndex_ = _getArgUint32(0x74);
     }
 
-    function gameData() external pure returns (GameType gameType_, Claim rootClaim_, bytes memory extraData_) {
-        gameType_ = GameType.wrap(1);
-        rootClaim_ = Claim.wrap(bytes32(0));
-        extraData_ = bytes("");
+    function extraData() public pure returns (bytes memory extraData_) {
+        extraData_ = _getArgBytes(0x54, 0x24);
     }
 
-    function wasRespectedGameTypeWhenCreated() external pure returns (bool) {
-        return false;
+    function gameData() external pure returns (GameType gameType_, Claim rootClaim_, bytes memory extraData_) {
+        gameType_ = gameType();
+        rootClaim_ = rootClaim();
+        extraData_ = extraData();
     }
 
-    /// @notice Legacy games expose `claimData(uint256)` rather than zero-arg version.
-    function claimData(uint256) external pure returns (bytes memory) {
-        return bytes("");
+    function wasRespectedGameTypeWhenCreated() external pure returns (bool) {
+        return true;
```

Comment:
Shouldn't we actually do something like the actual fault dispute game?
```
wasRespectedGameTypeWhenCreated =
    GameType.unwrap(ANCHOR_STATE_REGISTRY.respectedGameType()) == GameType.unwrap(GAME_TYPE);
```.

Then I think we should bake the anchor state registry into the mock dispute game as well.

Reply:
Good point. I actually thought about that. I don’t think it’s necessary, honestly. The thing is, since the factory is the creator, we can assume the mock games are already respected in our tests. So adding more would just bloat the mock.

Reply:
Yeah let's just keep it simple.